generator/mavparse.py

```diff
@@ -284,11 +291,22 @@ def char_data(data):
                 self.include.append(data)
 
         f = open(filename, mode='rb')
+        from lxml import etree
+        xmldocument = etree.parse(f)
+
+        if not self.with_wip:
+            # remove wip elements from parsing
+            wip_entry = xmldocument.findall('.//wip/..')
+            for entry in wip_entry:
+                entry.getparent().remove(entry)
+
         p = xml.parsers.expat.ParserCreate()
         p.StartElementHandler = start_element
         p.EndElementHandler = end_element
         p.CharacterDataHandler = char_data
-        p.ParseFile(f)
+        xmlstring = etree.tostring(xmldocument)
```

Comment:
FYI, - I use lxml.etree to remove the elements, then convert XML to a string that is then passed to the existing expat parser.